### PR TITLE
Add BSP for STM32F103RET6-OneNET-NBIoT Board .

### PR DESCRIPTION
**Pull request by DriftSnail**
ps: Already Completed the README.md of the BSP , and GPIO and UART Configurations.

## 拉取/合并请求描述：(PR description)

[
    Add BSP for STM32F103RET6-OneNET-NBIoT Board .
    By NiuQinghe , a member of RT-Thread Community group.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/stm32f103-onenet-nbiot/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+通过阅读快速上手章节开发者可以快速地上手该 BSP，将 RT-Thread 运行在开发板上。在进阶使用指南章节，将会介绍更多高级功能，帮助开发者利用 RT-Thread 驱动更多板载资源。
+
+## 开发板介绍
+
+OneNET NB-IoT V2.1 开发板采用高性能 ARM Cortex-M3 32 位微处理器 STM32F103RET6，搭载中移物联网公司自主研发的 M5310 NB 模组（内嵌 OneNET的接入协议：CoAP+LwM2M 协议），外围配备 SHT20 温湿度传感器、ADXL345
+三轴加速度传感器、BH1750FVI 光照传感器、BMP180 气压传感器、LED 指示灯、蜂鸣器及按键等丰富的外设资源。
+
+开发板实物如下图所示：
+
+![board](figures/board_detail.png)
```
Comment (SummerLife) [inline bsp/stm32/stm32f103-onenet-nbiot/README.md:22]: 建议完善一下图片，图片的宽一般设置为 800 像素

File: bsp/stm32/stm32f103-onenet-nbiot/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+| RTC            |     暂不支持     | 即将支持 |
+| IWDG | 暂不支持 | 即将支持 |
+| FLASH |     暂不支持     | 即将支持 |
+| **扩展模块**      | **支持情况** | **备注**                              |
+| 心率传感器 |   暂不支持   |                               |
+| OLED | 暂不支持 | |
+
+
+
+
```
Comment (SummerLife) [inline bsp/stm32/stm32f103-onenet-nbiot/README.md:59]: 删除多余空行

File: bsp/stm32/stm32f103-onenet-nbiot/board/linker_scripts/link.icf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+define block CSTACK    with alignment = 8, size = __ICFEDIT_size_cstack__   { };
+
+initialize by copy { readwrite };
+do not initialize  { section .noinit };
+
+place at address mem:__ICFEDIT_intvec_start__ { readonly section .intvec };
+
+place in ROM_region   { readonly };
+place in RAM_region   { readwrite, last block CSTACK};
```
Comment (SummerLife) [inline bsp/stm32/stm32f103-onenet-nbiot/board/linker_scripts/link.icf:28]: 需要在最后一行添加新的空行

File: bsp/stm32/stm32f103-onenet-nbiot/rtconfig.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    LFLAGS = ' --config "board/linker_scripts/link.icf"'
+    LFLAGS += ' --entry __iar_program_start'
+    
+    CXXFLAGS = CFLAGS
+
+    EXEC_PATH = EXEC_PATH + '/arm/bin/'
+    POST_ACTION = 'ielftool --bin $TARGET rtthread.bin'
+
+def dist_handle(BSP_ROOT):
```
Comment (SummerLife) [inline bsp/stm32/stm32f103-onenet-nbiot/rtconfig.py:145]: 近期 rt-thread 对 rtconfig.py 中的 dist_handle 做了一些完善，希望你能同步一下修改，请参考该链接 [link](https://github.com/RT-Thread/rt-thread/blob/b377859946a9ae30c902373d9773bed6ee319725/bsp/stm32/libraries/templates/stm32f10x/rtconfig.py#L145)

Comment (SummerLife) [reply bsp/stm32/stm32f103-onenet-nbiot/rtconfig.py:145]: @DriftSnail Hello 🌞

Comment (DriftSnail) [reply bsp/stm32/stm32f103-onenet-nbiot/rtconfig.py:145]: @SummerLife 
    rtconfig.py只有一个缩进的修改吧？这些刚刚都已经修改，需要重新PR吗？

Comment (SummerLife) [reply bsp/stm32/stm32f103-onenet-nbiot/rtconfig.py:145]:
```
def dist_handle(BSP_ROOT, dist_dir):
    import sys
    cwd_path = os.getcwd()
    sys.path.append(os.path.join(os.path.dirname(BSP_ROOT), 'tools'))
    from sdk_dist import dist_do_building
    dist_do_building(BSP_ROOT, dist_dir)
```

注意添加了 dist_dir 参数 @DriftSnail

**Comment by DriftSnail** [comment]
先前说的已经修改了，稍后我看看怎么合并一下这几个commit

**Comment by DriftSnail** [comment]
@SummerLife 我已经把几个commit合并了，麻烦看看是否可以拉取。 （操作不熟，还请见谅）

**Pull request merged**